### ISSUE TITLE
[Bug] Incorrect ngraph transformation from MatMul to FullyConnected 

### ISSUE THREAD

**Issue by evgeniya-egupova** [openvinotoolkit/openvino#15819]
##### System information (version)

- OpenVINO=> 2022.3.0
- Operating System / Platform => Ubuntu
- Compiler => :grey_question:
- Problem classification => MatMul ngraph transformation in CPU plugin

##### Detailed description
BatchMatMulV2 from TF with input shapes [18,N,1]x[18,1,80] has expected output shape [18,N,80]. It is represented as MatMul op in IR with correct shapes. 
However, during ngraph optimizations in the CPU plugin MatMul is transformed into FullyConnected op with the weights shape [1440,1], thus the shape of output tensor becomes [18,N,1440]. 
Problem is not reproduced using reference implementation.  

IR: 
![ir](https://user-images.githubusercontent.com/37940171/220100241-44218eb9-87aa-47e9-a9d6-5027836b3736.png)

Execution graph:  
![eg](https://user-images.githubusercontent.com/37940171/220100256-90ab856a-7e99-48a6-9d07-f5e9250164f4.png)

##### Steps to reproduce
IR: [broken_matmul.tar.gz](https://github.com/openvinotoolkit/openvino/files/10784108/broken_matmul.tar.gz)

```python
import numpy as np
from openvino.runtime import Core
core = Core()
model = core.read_model('broken_matmul.xml')
compiled_model = core.compile_model(model=model, device_name="CPU")
out_node = compiled_model.outputs[0]
out = compiled_model(np.ones((18, 1, 1)))[out_node]
print(f'Expected: {out_node.partial_shape}, got: {out.shape}')
```

**Comment by andrei-kochin** [comment]
@dmitry-gorokhov could you please assign someone to take a look?

**Comment by maxnick** [comment]
@v-Golubev , could you please take a look at this issue?

**Comment by v-Golubev** [comment]
A defect is in `MatMul`->`FullyConnected` conversion. The issue will be fixed in #15933

**Comment by evgeniya-egupova** [comment]
Works as expected. Closing the issue.
